docs/source/cuda-reference/kernel.rst

```diff
@@ -537,3 +537,35 @@ Timer Intrinsics
 
     Suspends the thread for a sleep duration approximately close to the delay
     ``ns``, specified in nanoseconds.
+
```

Comment:
It feels like "Types" should be a higher-level section all of its own:

```suggestion


Types
-----
```

Reply:
Should this be under kernel.rst or should it have its own page? Like host kernel, types has its own page: https://numba.pydata.org/numba-doc/dev/reference/types.html.

Reply:
I think perhaps starting a new page is a good idea - once that is done, we should at some point add reference to `float16` in it as well (but not for this PR / as part of this work).